bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -129,48 +127,54 @@ async function changeIDbyRealName(dataStore, client, timestamp) {
 
         do {
 
-            let user_ids = []
-            let indexUser = 0;
-            let text = channelMessages[indexMessages].text;
-            let searchText = text;
+            let userIds = []
+            let userNames = []
+            let userRegex = /(?:^|\W)<@(\w+)(?!\w)/g;
+            let match;
+            let replaceResult = [];
 
-            do {
+            let text = channelMessages[indexMessages].text;
 
-                const user_id = searchText.substring(searchText.indexOf('<@') + 2, searchText.indexOf('>'));
+            while (match = userRegex.exec(text)) {
+                userIds.push(match[1]);
```

Comment:
Wouldn't it be better to call `await client.fetch(`/users.info?user=${userId}`)` here?
You are filling two separate arrays(`userIds` and `userNames`) and then pair it later via `id`.
Imo it would be more readable if you put it into some structure together - i.e. an array of objects [{userId, userRealName}]

Reply:
Yes, I know.